osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for cron that performs the retention policy
+"""
+import datetime
+import logging
+import os
+import shutil
+import sys
+from datetime import date
+
+# Set up django configuration
+import django
+from django.db import DatabaseError
+from kobo.client.constants import FINISHED_STATES
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'osh.hub.settings'
+django.setup()
+
+from kobo.hub.models import Task  # noqa: E402
+
+from osh.hub.scan.models import RetentionPolicySetting  # noqa: E402
+from osh.hub.scan.models import TaskResultsRemoval  # noqa: E402
+
+logger = logging.getLogger("osh.hub.scripts.osh-retention")
+
+# The default value matches with the one defined on /hub/scan/models.py
+DEFAULT_RETENTION_DAYS = 30
+DEFAULT_CATEGORY = "Default Category"
+
+
+def delete_tasks_results(tasks_to_delete):
+    for task, retention_policy_setting in tasks_to_delete:
+        try:
+            path_to_delete = Task.get_task_dir(task.id)
+            # if folder exists, we delete it. if not, we mark it as deleted as
+            # the folder already doesn't exist
+            if os.path.isdir(path_to_delete):
+                shutil.rmtree(path_to_delete)
+                logger.debug('The following folder was deleted: %s', path_to_delete)
+        except OSError as e:
+            logger.error("Error deleting tasks folder: %s", e)
+            continue
+
+        # If the tasks results are deleted, we update the database field,
+        # so we don't try to delete them again.
+        TaskResultsRemoval.objects.create(task=task, reason=retention_policy_setting)
+
+
+# In order to identify personal tasks, we are using the owner field. If the owner doesn't
+# contain any / or @, we assume that it is a personal task.
+def identify_personal_task(owner):
+    # We could use regexps here if we want to make this more complex but for this specific case
+    # built-in string methods are enough, and we are more perfomance-oriented.
+    return "/" not in owner and "@" not in owner and "-" not in owner
+
+
+def main():
+    # Retention policy Task
+    logger.info("Starting Retention Policy Enforcement")
+
+    # Retrieval of all tasks ids that already have the retention policy applied
+    processed_tasks = TaskResultsRemoval.objects.values_list('task', flat=True)
```

Comment:
Do we need to fit the list of all previously processed tasks into the memory despite task ID is a foreign key in the table?  The list of processed tasks will grow indefinitely over time.  The time (and memory) taken by this script might depend on the number of tasks to be deleted but not on the number of tasks that have already been deleted in the previous runs.

Reply:
It is not necessary. I have modified the code to do only one query and we filter through the tasks that are not linked with the `TaskResultsRemoval` table. 
This also improves performance of the script and we should save memory